Add missing (polarization) terms in Vorticity equation

bshanahan opened the pull request:
Vorticity was missing the final terms in equation 4c (from the manual).  This PR tries to implement them.

bshanahan committed:
Add missing (polarization) terms in Vorticity equation

bshanahan commented:
Oops -- forgot to mention that I was simultaneously working on the 2D closures, which are included in this PR too.  I guess I can remove them if needed.

bshanahan committed:
Remove unused code from previous commit.

bshanahan committed:
Remove WIP hot ion sinks.

bendudson commented on hermes-2.cxx:
```diff
@@ -252,7 +252,8 @@ int Hermes::init(bool restarting) {
               .withDefault<bool>(true);
-  
+
+  OPTION(optsc, j_pol_terms, true);
```
Can we make the default `false` for now, at least until these new terms are working? I think they should be on by default eventually, but not yet...

bshanahan committed:
Set j_pol_terms to be false by default.

bendudson committed:
Modify terms, to use phi + Pi

Consistent with the Boussinesq approximation used in the calculation
of polarisation current.

bendudson reviewed (approved):
Thanks @bshanahan !
